src/game/clock.c

```diff
@@ -21,3 +23,42 @@ int32_t Clock_SyncTicks(int32_t target)
 {
     return S_Clock_SyncTicks(target);
 }
+
+int32_t Clock_GetYear()
+{
+    return S_Clock_GetYear();
+}
+
+int32_t Clock_GetMonth()
+{
+    return S_Clock_GetMonth();
+}
+
+int32_t Clock_GetDay()
+{
+    return S_Clock_GetDay();
+}
+
+int32_t Clock_GetHours()
+{
+    return S_Clock_GetHours();
+}
+
+int32_t Clock_GetMinutes()
+{
+    return S_Clock_GetMinutes();
+}
+
+int32_t Clock_GetSeconds()
+{
+    return S_Clock_GetSeconds();
+}
+
+char *Clock_GetDateTime(char *date_time)
+{
+    sprintf(
+        date_time, "%04d%02d%02d_%02d%02d%02d", Clock_GetYear(),
+        Clock_GetMonth(), Clock_GetDay(), Clock_GetHours(), Clock_GetMinutes(),
+        Clock_GetSeconds());
+    return date_time;
+}
```

Comment:
I don't think we use `Clock_GetMinutes` etc. publicly. Since I think it's safe to rely on `libc`, I'd inline all of these in this function, `Clock_GetDateTime` and not bother putting these in `s_clock.c`.

`Clock_GetDateTime` `sprintf`s the datetime and as such its signature is confusing. I'd either make it return a pointer to a static buffer (which is not thread-safe) or make it `void` to avoid confusion.

This file is missing a newline character at its end (GH highlights this with a stop sign). [See this to understand why it's needed](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline)

Reply:
Also consider allocating the memory for the return value here, making it `char *Clock_GetDateTime()` and make sure whoever uses it `Memory_Free`s its return value later.

Reply:
Ok I wasn't sure where to allocate the memory and made the clock functions for each hours, mins, secs, etc seprate in case they were ever needed elsewhere.

Reply:
The problem I had with this function was that it both returned a `char*` and used an out parameter essentially making it confusing about which is the real return value (and if the input argument being declared as `char*` instead of `const char*` is not a typo).